packages/sdk/src/swap/sdk.ts

```diff
@@ -159,46 +157,70 @@ export class SwapSDK {
     swapStatusRequest: SwapStatusRequest,
     options: RequestOptions = {},
   ): Promise<SwapStatusResponse> {
-    return ApiService.getStatus(this.baseUrl, swapStatusRequest, options);
+    return ApiService.getStatus(
+      this.options.backendUrl,
+      swapStatusRequest,
+      options,
+    );
   }
 
   async executeSwap(
     params: ExecuteSwapParams,
-    txOpts: TransactionOptions = {},
+    txOpts: TransactionOptions & { signer?: Signer } = {},
   ): Promise<TransactionHash> {
     const { srcChain, srcAsset, amount } = params;
 
+    const { signer: optsSigner, ...remainingTxOpts } = txOpts;
+    const signer = optsSigner ?? this.options.signer;
+    assert(signer, 'No signer provided');
+    assert(signer.provider, 'Signer has no provider');
+
+    const sourceChainId = getEvmChainId(params.srcChain, this.options.network);
```

Comment:
check that given signer is on the correct chain for the swap, this will be important once we support multiple evm chains